geom/FieldBinaryOperatorTemplate.H

```diff
@@ -0,0 +1,51 @@
+#define _NAME field
+#include "function_overload.macro"
+
+   subroutine _SUB(field_a,field_b,undef,rc)
+      type(ESMF_Field), intent(inout) :: field_a
+      type(ESMF_Field), intent(inout) :: field_b
+      real, optional, intent(in) :: undef
+      integer, optional, intent(out) :: rc
```

Comment:
I'm confused.   A binary operator should have 3 arguments:    `C = A .op. B`

Reply:
@tclune, what do you want me to name this. This is the place I put +-/*, I can call it whatever you want. What name collective describes those 4 operations?